src/main/java/graphql/GraphQLArgumentInstrumentation.java

```diff
@@ -0,0 +1,13 @@
+package graphql;
+
+import graphql.schema.GraphQLArgument;
+
+/**
+ * You can add an instance of this class to your {@link GraphQLArgument}.
+ * The instrumentValue method will be called every time a argument value is being resolved in the {@link graphql.execution.ValuesResolver}.
+ * For example, if the client passes an argument value "hello" to an argument in your field, you can write an instrumentValue implementation that transforms it to upper case letters.
+ * Then, the data fetcher environment will contain a value of "HELLO".
+ */
+public interface GraphQLArgumentInstrumentation {
```

Comment:
What are people's opinion of having this just be extending a `Function`. For example:

```
public interface GraphQLArgumentMapper extends Function<Object, Object> {

}
```
The one positive for this is it will allow for the consumer to apply chaining etc without having to build our own `ChainingGraphQLArgumentMapper`.  For example given the following mappers:

```
   public interface GraphQLArgumentMapper extends Function<Object, Object> {
        
    }
    
    public class UpperCaseGraphQLArgumentMapper implements GraphQLArgumentMapper {
        @Override
        public Object apply(final Object o) {
            if (o instanceof String) {
                return ((String) o).toUpperCase();
            }
            
            return o;
        }
    }

    public class StringPrefixGraphQLArgumentMapper implements GraphQLArgumentMapper {
        @Override
        public Object apply(final Object o) {
            if (o instanceof String) {
                return "prefix" + o;
            }

            return o;
        }
    }
```

then as a consumer you can then chain them like so:

``` 
        GraphQLArgumentMapper f = new UpperCaseGraphQLArgumentMapper();
        GraphQLArgumentMapper g = new StringPrefixGraphQLArgumentMapper();

        f.andThen(g)
                .apply("test");
```

Reply:
It's a nice idea. I think in this case the whole `GraphQLArgumentMapper` class is redundant and the type of the mapper can be just `Function`